pytorch_lightning/trainer/trainer_io.py

```diff
@@ -93,6 +96,18 @@ def restore_state_if_checkpoint_exists(self, model):
 
         return did_restore
 
+    def restore_state_from_checkpoint(self, checkpoint_path):
+        did_restore = False
+
+        checkpoint_path = Path(checkpoint_path)
+        if not checkpoint_path.exists():
```

Comment:
does not work for `str` as it is defined `:param resume_from_checkpoint: str or os.PathLike object.`

Reply:
checkpoint_path is what torch.load expects as input. It can be file-like object or str containing a file name.
https://pytorch.org/docs/stable/torch.html?highlight=torch%20load#torch.load